contextual_orchestrator/cost_router.py

```
@@ -92,6 +89,43 @@ def __init__(
+                matching = [agent for agent in candidates if agent.model == model]
+                if not matching:
+                    raise ValueError(f"embedding model {model!r} is not configured in the embedding agent pool")
+                indexes = [index for index, request in enumerate(requests) if request.model == model]
+                client = getattr(self.orchestrator, "client", None)
+                embed_many = getattr(client, "embed_many", None)
+                if not callable(embed_many):
+                    raise RuntimeError("configured embedding agent has no provider embedding client")
+                batch_vectors = embed_many(matching[0], [requests[index].input_text for index in indexes])
+                if len(batch_vectors) != len(indexes):
+                    raise RuntimeError("provider returned an incomplete embedding vector batch")
+                for index, vector in zip(indexes, batch_vectors):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)